contracts/src/gateway/router/CheckpointingFacet.sol

```diff
@@ -52,9 +52,42 @@ contract CheckpointingFacet is GatewayActorModifiers {
         bytes32 membershipRootHash,
         uint256 membershipWeight
     ) external systemActorOnly {
-        if (checkpoint.blockHeight % s.bottomUpCheckPeriod != 0) {
-            revert InvalidCheckpointEpoch();
-        }
+        // TODO: Discussion:
+        //
+        // This check is removed:
+        //
+        // if (checkpoint.blockHeight % s.bottomUpCheckPeriod != 0) {
+        //    revert InvalidCheckpointEpoch();
+        // }
+        //
+        // First of all, the check is buggy in the first place as it does not support
+        // a full message batch before a checkpoint period is reached.
+        // Then, `prevBottomUpCheckpointHeight` is introduced, as long as BU checkpoints
+        // can form a chain, it should be fine.
+        //
+        // Instead, one can add the following check:
+        //
+        // if (checkpoint.prevHeight != s.prevBottomUpCheckpointHeight) {
+        //     revert InvalidCheckpointEpoch();
+        // }
+        //
+        // However, we are restricting to ONLY sequential creation
+        // of bottom up checkpoints. One can only submit the next one if the current
+        // checkpoint quorum is reached. This would slow down the creation process.
+        //
+        // Or we can enforce:
+        //
+        // checkpoint.msgs.length != 0 || s.prevConfigNumber < checkpoint.nextConfigurationNumber
+        //
+        // so that no empty checkpoint will be created. The drawback is we might have long block intervals with
+        // no checkpoints at all. We can get around this with some `maxBUGap` or something along this line, so that
+        // when no bottom up checkpoints, we dont allow long gap where there is no bottom up checkpoints. This also
```

Comment:
Yes I think the formula we discussed in the ticket you created included this as an option. 
Checkpointing does have value on its own, so I wouldn't come up with a rule that says no, we don't do any empty checkpoints, period. This is a good time to make it configurable.

Can you include a link to the ticket with the formula?

Reply:
is there a formula? I was not aware of this.